django/db/backends/base/operations.py

```diff
@@ -655,3 +658,17 @@ def window_frame_rows_start_end(self, start=None, end=None):
 
     def window_frame_range_start_end(self, start=None, end=None):
         return self.window_frame_rows_start_end(start, end)
+
+    def explain_query_prefix(self, format=None, **options):
+        if not self.explain_prefix:
+            raise NotSupportedError('This backend does not support explaining query execution.')
+
+        if format:
+            supported_formats = self.connection.features.supported_explain_formats
+            if format.upper() not in supported_formats:
+                msg = '{0} is not a recognised format. '.format(format.upper())
```

Comment:
In the current implementation if ``supported_formats`` evaluates to ``False``, then there would be a trailing space at the end of the error message. Probably it would be better to remove it from this string and add it to the string below:

```
msg += ' Allowed formats: {0}'.format(', '.join(supported_formats))
```

Reply:
Done, thanks!